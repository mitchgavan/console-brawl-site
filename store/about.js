export default {
  title: 'The App',
  tagline: 'A side-project built with React Native',
  body: [
    'Console Brawl is a React Native app currently available for iOS and Android. This is a side project I\'ve been working on for a while now; from initial concept, through to design and development, and finally deployment to the app stores. This is my first native app build, and as you can imagine I learnt a lot throughout the process.',
    'The basic app concept is to pit two video games consoles against each other and find out which console you prefer. The user will then be presented with a game from each platform, and choose which one they prefer. After a number of rounds, their preferred platform is calculated based on the number of games they liked that are only available on that platform.',
    'It\'s a pretty simple idea, but something that I thought would be kind of fun! Maybe it\'s because I grew up in the days of Sega versus Nintendo, and then Sony versus Nintendo. Where it just seemed like everybody was loyal to one or the other and the school yard arguments over which console was better would rage. So in Console Brawl, you can select from any console from the 8-bit era right up to the current generation, and see which one you\'d prefer.',
    'This is possible thanks to the Giant Bomb API, which is an open source games database, with data on pretty much every game ever released.',
  ],
};
