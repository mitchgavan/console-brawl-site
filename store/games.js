import axios from 'axios';
import base64 from 'base-64';
import config from '../config';

const headers = {
  'Accept': 'application/json',
  'Content-Type': 'application/json',
  'Authorization': 'Basic bWl0Y2hnMjM6WUtzaDNjdDZUWTdYZTJEbg==',
};

function getRecentPlatformsGames() {
  return axios.get(`${config.API_URL_RECENT}/games`, { headers })
  .then((response) => {
    console.log(response);
  })
  .catch((error) => {
    console.log(error);
  });
}

export { getRecentPlatformsGames };
