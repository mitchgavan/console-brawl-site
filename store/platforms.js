export const platformsA = [
  'master-system',
  'super-nintendo',
  'sega-saturn',
  'nintendo-64',
  'xbox',
  'gamecube',
  'xbox-360',
  'playstation-4',
];

export const platformsB = [
  'nintendo',
  'sega-mega-drive',
  'playstation-1',
  'sega-dreamcast',
  'playstation-2',
  'wii',
  'playstation-3',
  'xbox-one',
];

export const developers = [
  'Microsoft',
  'Nintendo',
  'Sega',
  'Sony',
];

export const platformDetails = [
  {
    name: 'Nintendo (NES)',
    slug: 'nintendo',
    released: 'July 15, 1983',
    developer: 'Nintendo',
    description: 'The 8-bit console helped reignite the video games industry in North America after the crash of 1983. In 2009 IGN named it the single greatest video games console in history. Oh and Super Mario Bros changed the game as we knew it.',
  },
  {
    name: 'Master System',
    slug: 'master-system',
    released: 'October 20, 1985',
    developer: 'Sega',
    description: 'Let the rivalry begin, this was Sega\'s direct competitor to the NES. Although it failed to compete with it in sales in the US and Japanese markets, it had a lot more success in PAL territories. Some stand out titles include Alex the Kidd and Wonder Boy.',
  },
  {
    name: 'Super NES',
    slug: 'super-nintendo',
    released: 'November 21, 1990',
    developer: 'Nintendo',
    description: 'The successor to the NES would face fierce competition from the Sega Mega Drive, but ultimately was the best selling console of the 16-bit generation. Super Mario World, Zelda: A Link to the Past - just a couple of games that made this an all time classic.',
  },
  {
    name: 'Mega Drive',
    slug: 'sega-mega-drive',
    released: 'October 29, 1988',
    developer: 'Sega',
    description: 'If you were a kid in the 16-bit era, it was a common narrative to hear that Sega\'s console was the cooler console. Shit, their Mortal Kombat had "real" red blood while the Super NES had that fake green stuff. And this is also the console that Sonic the Hedgehog was first introduced.',
  },
  {
    name: 'Playstation 1',
    slug: 'playstation-1',
    released: 'December 3, 1994',
    developer: 'Sony',
    description: 'Welcome Sony, the industry will never be the same. A focus on 3D polygon graphics and Sony\'s approach to third party developers helped the Playstation become a market success. It introduced franchises including Gran Turismo, Crash Bandicoot, Metal Gear Solid and Tekken',
  },
  {
    name: 'Sega Saturn',
    slug: 'sega-saturn',
    released: 'November 22, 1994',
    developer: 'Sega',
    description: 'Complex hardware design and lack of third party support lead to the Saturn\'s commercial failure. But it\'s remembered these days for several great games like Night\'s into Dreams, Panzer Dragoon and also a healthy library of shmups - including the classic Radiant Silvergun.',
  },
  {
    name: 'Nintendo 64',
    slug: 'nintendo-64',
    released: 'June 23, 1996',
    developer: 'Nintendo',
    description: 'The N64 was the last to the party in this generation, but it boasted the most impressive specs. Super Mario 64, Ocarina of Time, Goldenye - with some of the greatest and most influential games of all time the Nintendo 64 remains one of the most recoginised consoles in history.',
  },
  {
    name: 'Sega Dreamcast',
    slug: 'sega-dreamcast',
    released: 'November 27, 1998',
    developer: 'Sega',
    description: 'The Dreamcast enjoyed a successful launch, but that soon slowed down and ended after a short lifespan. Lack of third party support was one of the main reasons for this. Nevertheless, it produced some great games including Shenmue, Jet Set Radio and Crazy Taxi. Along with a host of arcade classics.',
  },
  {
    name: 'Playstation 2',
    slug: 'playstation-2',
    released: 'March 4, 2000',
    developer: 'Sony',
    description: 'One of the greatest.',
  },
  {
    name: 'Xbox',
    slug: 'xbox',
    released: 'November 15, 2001',
    developer: 'Microsoft',
    description: 'Microsoft throws it\'s cards on the table.',
  },
  {
    name: 'Gamecube',
    slug: 'gamecube',
    released: 'September 14, 2001',
    developer: 'Nintendo',
    description: 'A little bit different.',
  },
  {
    name: 'Xbox 360',
    slug: 'xbox-360',
    released: 'November 22, 2005',
    developer: 'Microsoft',
    description: 'Microsoft proves that they\'re here to stay',
  },
  {
    name: 'Wii',
    slug: 'wii',
    released: 'November 19, 2006',
    developer: 'Nintendo',
    description: 'Nintendo goes in a totally different direction.',
  },
  {
    name: 'Playstation 3',
    slug: 'playstation-3',
    released: 'November 11, 2006',
    developer: 'Sony',
    description: 'Sony continues it\'s dominance.',
  },
  {
    name: 'Wii U',
    slug: 'wii-u',
    released: 'November 18, 2012',
    developer: 'Nintendo',
    description: 'The new Wii, some cool games, but ultimately a failure.',
  },
  {
    name: 'Xbox One',
    slug: 'xbox-one',
    released: 'November 22, 2013',
    developer: 'Microsoft',
    description: 'Another great console.',
  },
  {
    name: 'Playstation 4',
    slug: 'playstation-4',
    released: 'November 15, 2013',
    developer: 'Sony',
    description: 'The latest and greatest, with VR.',
  },
];
