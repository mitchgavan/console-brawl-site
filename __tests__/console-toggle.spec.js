import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ConsoleToggle from '../components/console-list/console-toggle';
import { developers } from '../store/platforms';

test('It renders correctly', () => {
  let tree = shallow(
    <ConsoleToggle
      developers={developers}
      toggle={() => false}
      toggledOn="All"
    />,
  );
  expect(toJson(tree)).toMatchSnapshot();

  tree = shallow(
    <ConsoleToggle
      developers={developers}
      toggle={() => false}
      toggledOn="Nintendo"
    />,
  );
  expect(toJson(tree)).toMatchSnapshot();
});
