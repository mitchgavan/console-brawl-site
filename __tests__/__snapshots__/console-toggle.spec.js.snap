// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It renders correctly 1`] = `
<div
  className="root"
  data-jsx={3417253327}
>
  <div
    className="toggle"
    data-jsx={3417253327}
  >
    <button
      className="toggle__button is-active"
      data-jsx={3417253327}
      onClick={[Function]}
    >
      All
    </button>
    <button
      className="toggle__button false"
      data-jsx={3417253327}
      onClick={[Function]}
    >
      Microsoft
    </button>
    <button
      className="toggle__button false"
      data-jsx={3417253327}
      onClick={[Function]}
    >
      Nintendo
    </button>
    <button
      className="toggle__button false"
      data-jsx={3417253327}
      onClick={[Function]}
    >
      Sega
    </button>
    <button
      className="toggle__button false"
      data-jsx={3417253327}
      onClick={[Function]}
    >
      Sony
    </button>
  </div>
  <_class
    css=".root[data-jsx=\\"3417253327\\"] {padding: 0;text-align: center;}.toggle[data-jsx=\\"3417253327\\"] {position: relative;display:-webkit-flex; display:flex;margin-bottom: 2rem;padding: 14px;background-color: #121212;z-index: 1;overflow: auto;}.toggle__button[data-jsx=\\"3417253327\\"] {padding: 7px 20px;border-radius: 100em;color: #fafafa;}.toggle__button.is-active[data-jsx=\\"3417253327\\"] {background-color: #3b3b3b;}@media screen and (min-width: 769px) {.root[data-jsx=\\"3417253327\\"] {margin-top: -30px;padding: 0;}.toggle[data-jsx=\\"3417253327\\"] {display:-webkit-inline-flex; display:inline-flex;justify-content: center;}}"
    styleId={3417253327}
  />
</div>
`;

exports[`It renders correctly 2`] = `
<div
  className="root"
  data-jsx={3417253327}
>
  <div
    className="toggle"
    data-jsx={3417253327}
  >
    <button
      className="toggle__button false"
      data-jsx={3417253327}
      onClick={[Function]}
    >
      All
    </button>
    <button
      className="toggle__button false"
      data-jsx={3417253327}
      onClick={[Function]}
    >
      Microsoft
    </button>
    <button
      className="toggle__button is-active"
      data-jsx={3417253327}
      onClick={[Function]}
    >
      Nintendo
    </button>
    <button
      className="toggle__button false"
      data-jsx={3417253327}
      onClick={[Function]}
    >
      Sega
    </button>
    <button
      className="toggle__button false"
      data-jsx={3417253327}
      onClick={[Function]}
    >
      Sony
    </button>
  </div>
  <_class
    css=".root[data-jsx=\\"3417253327\\"] {padding: 0;text-align: center;}.toggle[data-jsx=\\"3417253327\\"] {position: relative;display:-webkit-flex; display:flex;margin-bottom: 2rem;padding: 14px;background-color: #121212;z-index: 1;overflow: auto;}.toggle__button[data-jsx=\\"3417253327\\"] {padding: 7px 20px;border-radius: 100em;color: #fafafa;}.toggle__button.is-active[data-jsx=\\"3417253327\\"] {background-color: #3b3b3b;}@media screen and (min-width: 769px) {.root[data-jsx=\\"3417253327\\"] {margin-top: -30px;padding: 0;}.toggle[data-jsx=\\"3417253327\\"] {display:-webkit-inline-flex; display:inline-flex;justify-content: center;}}"
    styleId={3417253327}
  />
</div>
`;
